src/components/Map/LeafletMap.js

```diff
@@ -1,83 +1,97 @@
-  const extent = regions[1].mapProperties.extent // conus - TODO: I hate this how can I fix this?
+  const dispatch = useDispatch();
+  const selectedRegion = useSelector(selectedRegionSelector);
+  const zoom = useSelector(selecteZoomSelector);
+  const center = useSelector(selecteCenterSelector);
+  const extent = regions[1].mapProperties.extent; // conus - TODO: I hate this how can I fix this?
 
   const classes = useStyles();
 
   const handleRegionSelectChange = (event) => {
     // Update map with new center and zoom
-    map.setView(regions[event.target.value].mapProperties.center, 
-                regions[event.target.value].mapProperties.zoom)
-    
+    map.setView(
+      regions[event.target.value].mapProperties.center,
+      regions[event.target.value].mapProperties.zoom
+    );
+
     // Update redux store with new region, zoom, and center
-    dispatch(changeRegionValue(event.target.value))
-    dispatch(changeZoom(regions[event.target.value].mapProperties.zoom))
-    dispatch(changeCenter(regions[event.target.value].mapProperties.center))
-  }
+    dispatch(changeRegionValue(event.target.value));
+    dispatch(changeZoom(regions[event.target.value].mapProperties.zoom));
+    dispatch(changeCenter(regions[event.target.value].mapProperties.center));
+  };
 
   const displayMap = useMemo(
     () => (
-      <MapContainer className = {classes.leafletContainer} 
-        center={center} 
-        zoom={zoom} 
-        scrollWheelZoom={true}
-        bounds={extent}
-        whenCreated={setMap}
-      >
-        <link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css"
-          integrity="sha512-xodZBNTC5n17Xt2atTPuE1HxjVMSvLVW9ocqUKLsCC5CXdbqCmblAshOMAS6/keqq/sMZMZ19scR4PsZChSR7A=="
-          crossOrigin=""/>
-        <link rel="stylesheet" href="https://unpkg.com/leaflet-draw@latest/dist/leaflet.draw-src.css" />
+      <MapContainer className = {classes.leafletContainer}
+          center={center}
+          zoom={zoom}
+          scrollWheelZoom={true}
+          bounds={extent}
+          whenCreated={setMap}
+          >
+            <link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css"
+            integrity="sha512-xodZBNTC5n17Xt2atTPuE1HxjVMSvLVW9ocqUKLsCC5CXdbqCmblAshOMAS6/keqq/sMZMZ19scR4PsZChSR7A=="
+            crossOrigin=""/>
+          <link rel="stylesheet" href="https://unpkg.com/leaflet-draw@latest/dist/leaflet.draw-src.css" />
         <TileLayer
           attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
           url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
         />
       </MapContainer>
     ),
-    [],
-  )
+    [center, classes.leafletContainer, extent, zoom]
+  );
+
   return (
     <div>
-      {map ? <BasicSelect defaultValue={selectedRegion} values={regions} onChange={handleRegionSelectChange}/>: null}
+      {map ? <BasicSelect
```

Comment:
lines 89-92 Is this a result of the linter too?

Reply:
Yes, this is all the Airbnb linter. I feel like Airbnb is a popular set of lint rules. BTW I do not auto-fix anything because I want to learn what's wrong and why. IMHO 2 spaces are not enough to visually see a difference.